MetaMorpheus/EngineLayer/Util/DictionaryExtensions.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Nett;
+using Newtonsoft.Json.Linq;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Numerics;
+
+namespace EngineLayer;
+
+public static class DictionaryExtensions
+{
+    private static readonly object AddOrCreateLock = new object();
+
```

Comment:
I don't like having a static lock for an extension methods. Multiple dictionaries could end up using the extension method, and you want a one-to-one mapping between locks and dictionaries